packages/vm/src/vm.ts

```diff
@@ -31,7 +31,10 @@ import type { StateManager } from '@ethereumjs/statemanager'
  *
  * This class is an AsyncEventEmitter, please consult the README to learn how to use it.
  */
-export class VM extends AsyncEventEmitter<VMEvents> {
+export class VM<
+  EVMType extends EVMInterface = EVMInterface,
+  EEIType extends EEIInterface = EEIInterface
```

Comment:
I don't think this is correct; I think the default types are `EVM` and `EEI`, right?

Reply:
We are (or: @jochem-brouwer is) testing out a new approach here at #2208 which might be able to leverage the resulting type problems by preventing this type split in the first place. We'll see. 🙂